src/views/product-integrations-landing/components/paginated-integrations-list/index.tsx

```diff
@@ -0,0 +1,196 @@
+import { IconArrowLeft16 } from '@hashicorp/flight-icons/svg-react/arrow-left-16'
+import { IconArrowRight16 } from '@hashicorp/flight-icons/svg-react/arrow-right-16'
+import classNames from 'classnames'
+import { Integration } from 'lib/integrations-api-client/integration'
+import { useEffect, useState } from 'react'
+import IntegrationsList from '../integrations-list'
+import s from './style.module.css'
+
+interface PaginatedIntegrationsListProps {
+	integrations: Array<Integration>
+	itemsPerPage?: number
+}
+
+export default function PaginatedIntegrationsList({
+	integrations,
+	itemsPerPage = 8,
+}: PaginatedIntegrationsListProps) {
+	// Sort integrations alphabetically
+	const sortedIntegrations = integrations.sort(
+		(a: Integration, b: Integration): number => {
+			if (a.name < b.name) {
+				return -1
+			} else if (a.name > b.name) {
+				return 1
+			} else {
+				return 0
+			}
+		}
+	)
+
+	// Keep track of the current page & Integrations to display
+	const [currentPage, setCurrentPage] = useState(1)
+	const currentPageIntegrations = sortedIntegrations.slice(
+		(currentPage - 1) * itemsPerPage,
+		currentPage * itemsPerPage
+	)
+
+	// If our integrations change, set the page back to 1
+	useEffect(() => {
+		setCurrentPage(1)
+	}, [sortedIntegrations])
+
+	return (
+		<div className={s.paginatedIntegrationsList}>
+			<IntegrationsList integrations={currentPageIntegrations} />
+			{/* It's possible that all integrations display on a single page
+      in that case, just don't even show the paginator. */}
+			{currentPageIntegrations.length < integrations.length && (
+				<div className={s.paginatorWrapper}>
+					<Paginator
+						numberOfPages={Math.ceil(integrations.length / itemsPerPage)}
+						currentPage={currentPage}
+						onPageClicked={(page: number) => {
+							setCurrentPage(page)
+						}}
+					/>
+				</div>
+			)}
+		</div>
+	)
+}
+
+// =======================
+
+const COLLAPSED = '...'
+
+function paginatedArray(
```

Comment:
Like `unflattenVariables`, i probably will gloss over this and trust that it works :D 

A quick top-level comment for expected output could be useful here 🙏

Reply:
Good idea; I'll add a comment!